### ISSUE TITLE
Reading from /dev/input/event0 does not work anymore

### ISSUE THREAD

**Issue by joerg-krause** [luvit/luvit#950]
Commit https://github.com/luvit/luvit/commit/edf48cf3093a574805595e612680c006a6cdf741 was added by @Miigon to fix https://github.com/luvit/luvit/issues/940, but unfortunately breaks reading from `/dev/input/event0`, which was working fine before.

The following code snippet does not work anymore:

``` Lua
local fs = require("fs")
local Emitter = require("core").Emitter
local KeyEventListener = Emitter:extend() -- class table

function KeyEventListener:initialize(id)
  self.id = id
  self:wrap("onOpen")
  self:wrap("onRead")
  fs.open("/dev/input/event" .. id, "r", "0644", self.onOpen)
end

function KeyEventListener:onOpen(fd)
  self.fd = fd
  self:startRead()
end

function KeyEventListener:startRead()
  fs.read(self.fd, ffi.sizeof("struct input_event"), nil, self.onRead)
end

function KeyEventListener:onRead(data)
  p(data)
  if self.fd then self:startRead() end
end

function KeyEventListener:close(callback)
  local fd = self.fd
  self.fd = nil
  fs.close(fd, callback)
end

return {
  KeyEventListener = KeyEventListener
}
```

The callback function `onRead()` passed to `fs.read()` is never called on an input event. Reverting this commit helps.

**Comment by creationix** [comment]
:(

**Comment by Miigon** [comment]
Since I just replaced `fs.read` with `fs.readSync`,maybe these sockets don't support `fs.readSync`?

**Comment by creationix** [comment]
@Miigon I think the problem is you turned a non-blocking function into a blocking function and so it no longer invokes the callback.

I don't know this API very well, but as I commented in your commit, I think it's supposed to be non-blocking.

**Comment by joerg-krause** [comment]
Fixed in version 2.14.1.
